src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/NoEnumTrailingCommaCheck.java

```diff
@@ -0,0 +1,129 @@
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+/**
+ * <p>
+ *  Checks that enum definition does not contain a trailing comma.
+ *  Rationale: JLS allows trailing commas in arrays and enumerations, but does not allow
+ *  them in other locations. To unify the coding style, the use of trailing commas should
+ *  be prohibited.
+ * </p>
+ * <pre>
+ * enum Foo1 {
+ *   FOO,
+ *   BAR;
+ * }
+ * </pre>
+ * <p>
+ *  The check demands that there should not be any comma after last constant in
+ *  enum definition.
+ * </p>
+ * <pre>
+ * enum Foo1 {
+ *   FOO,
+ *   BAR, //violation
+ * }
+ * </pre>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;NoEnumTrailingComma&quot;/&gt;
+ * </pre>
+ * <p>
+ * Which results in the following violations:
+ * </p>
+ * <pre>
+ * enum Foo1 {
+ *   FOO,
+ *   BAR; //OK
+ * }
+ * enum Foo2 {
+ *   FOO,
+ *   BAR //OK
+ * }
+ * enum Foo3 {
+ *   FOO,
+ *   BAR, //violation
+ * }
+ * enum Foo4 {
+ *   FOO, // OK
+ *   BAR, // violation
+ *   ;
+ * }
+ * enum Foo5 {
+ *   FOO, // OK
+ *   BAR,; // violation
+ * }
+ * enum Foo6 { FOO, BAR,; } // violation
+ * enum Foo7 { FOO, BAR, } // violation
+ * enum Foo8 {
+ *   FOO, // OK
+ *   BAR // OK
+ *   ;
+ * }
+ * enum Foo9 { FOO, BAR; } // OK
+ * enum Foo10 { FOO, BAR } // OK
+ * </pre>
+ *
+ * @since 8.29
+ */
+@StatelessCheck
+public class NoEnumTrailingCommaCheck extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "no.enum.trailing.comma";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.ENUM_DEF};
+    }
+
+    @Override
+    public void visitToken(DetailAST detailAST) {
+        final DetailAST objBlock = detailAST.findFirstToken(TokenTypes.OBJBLOCK);
+        TokenUtil.forEachChild(objBlock, TokenTypes.COMMA, comma -> {
+            final int type = comma.getNextSibling().getType();
+            if (type != TokenTypes.VARIABLE_DEF && type != TokenTypes.ENUM_CONSTANT_DEF) {
```

Comment:
I don't understand why we are looking at a VARIABLE_DEF? Shouldn't everything in the start of the enum be the `ENUM_CONSTANT_DEF`? If we hit a variable def, that means we are going past the end `;` for the list of enumerations.

Reply:
It is a good catch! The iteration should stop at `;` , this will solve the regression with `VARIABLE_DEF`.

Reply:
>this will solve the regression with VARIABLE_DEF

Do you see it at report ?

Reply:
I don't think this will be required after making the change suggested in [#7404Comment](https://github.com/checkstyle/checkstyle/pull/7404#discussion_r362119936)